### PR TITLE
Only close the span after all writing has finished so that it can sti…

### PR DESCRIPTION
**Pull request by bcluap**
Addresses one part of #16106

### CODE REVIEW

File: extensions/smallrye-opentracing/runtime/pom.xml [hunk 1]
```diff
@@ -110,6 +110,19 @@
                     </annotationProcessorPaths>
                 </configuration>
             </plugin>
+            <!-- The filter classes needs to be indexed -->
+            <plugin>
```
Comment (geoand) [inline extensions/smallrye-opentracing/runtime/pom.xml:114]: I think we can forgo this by generating a JaxrsProviderBuildItem with the provider you added

**Review by geoand — COMMENTED** [review]
Nice work!

I added a comment about the registration of the provider

File: extensions/smallrye-opentracing/runtime/pom.xml (hunk 1 above)
Comment (bcluap) [reply extensions/smallrye-opentracing/runtime/pom.xml:114]: Any examples of that I can look at? I'm still a newborn when it comes to extensions :-)

Comment (geoand) [reply extensions/smallrye-opentracing/runtime/pom.xml:114]: Any of the usages of https://github.com/quarkusio/quarkus/blob/main/extensions/resteasy-classic/resteasy-common/spi/src/main/java/io/quarkus/resteasy/common/spi/ResteasyJaxrsProviderBuildItem.java should be just fine 🙂

Comment (bcluap) [reply extensions/smallrye-opentracing/runtime/pom.xml:114]: Looking at examples I think this should work:
```
} else if (capabilities.isPresent(Capability.RESTEASY_REACTIVE)) {
            customResponseFilters.produce(new CustomContainerResponseFilterBuildItem(
                    QuarkusSmallRyeTracingStandaloneContainerResponseFilter.class.getName()));
            dynamicFeatures.produce(new DynamicFeatureBuildItem(QuarkusSmallRyeTracingDynamicFeature.class.getName()));
// new...
            providers.produce(
                    new ResteasyJaxrsProviderBuildItem(
                            QuarkusSmallRyeTracingStandaloneContainerResponseFilter.class.getName()));
}
```

But alas its not being registered

Comment (geoand) [reply extensions/smallrye-opentracing/runtime/pom.xml:114]: Ah darn. Sorry, the build item I added above is for RESTEasy Classic

Comment (geoand) [reply extensions/smallrye-opentracing/runtime/pom.xml:114]: https://github.com/quarkusio/quarkus/blob/main/extensions/resteasy-reactive/quarkus-resteasy-reactive-common/spi-deployment/src/main/java/io/quarkus/resteasy/reactive/spi/ContainerRequestFilterBuildItem.java should be the one to uss

Comment (bcluap) [reply extensions/smallrye-opentracing/runtime/pom.xml:114]: How about this one:
writerInterceptors.produce(
                    new WriterInterceptorBuildItem.Builder(
                            QuarkusSmallRyeTracingStandaloneContainerResponseFilter.class.getName()).build());

It works btw

Comment (geoand) [reply extensions/smallrye-opentracing/runtime/pom.xml:114]: Ah, it's an interceptor! So yeah, that should be the one to use!

**Review by geoand — COMMENTED** [review]
Can you please squash the commits?

Thanks

**Pull request closed without merging**